Subject: [PATCH 08/19] autofs-5.1.8 - fix loop under run in
 cache_get_offset_parent()
From:   Ian Kent <raven@themaw.net>
From: Frank Sorenson <sorenson@redhat.com>

To avoid reading memory outside of the the string
allocated for parent, tail needs to stop when it
reaches or passes parent, even if it doesn't
actually equal parent.

Signed-off-by: Frank Sorenson <sorenson@redhat.com>
---
 lib/cache.c |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

Origin: upstream, https://www.spinics.net/lists/autofs/msg02432.html
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/autofs/+bug/1955851
Last-Update: 2022-01-20
Backport note: dropped the CHANGELOG patch hunk
--- a/lib/cache.c
+++ b/lib/cache.c
@@ -710,7 +710,7 @@
 		*tail = 0;
 
 		tail--;
-		if (tail == parent)
+		if (tail <= parent)
 			break;
 
 		me = cache_lookup_distinct(mc, parent);
